### PR TITLE
Elytron Phase I - Continuation

### PR DESCRIPTION
**Pull request by darranl**
:exclamation: Commits from this branch are used in numerous other topic branches so please do not cherry-pick or rebase. :exclamation:

This pull request is a continuation of phase I of the Elytron pull requests, incorporating fixes for issues discovered reviewing the last pull request and subsequent bug reports raised.

For this pull request the following issues can be marked as resolved after merging: -

https://issues.jboss.org/browse/WFLY-7044
https://issues.jboss.org/browse/WFLY-7045
https://issues.jboss.org/browse/WFLY-7047
https://issues.jboss.org/browse/WFLY-7051
https://issues.jboss.org/browse/WFLY-7039
https://issues.jboss.org/browse/WFLY-7061

### CODE REVIEW

**Comment by kabir** [comment]
Retest this please

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpsListenerResourceDefinition.java [hunk 1]
```diff
@@ -60,14 +60,14 @@
             .setCapabilityReference(SSL_CONTEXT_CAPABILITY, LISTENER_CAPABILITY_NAME, true)
             .setFlags(AttributeAccess.Flag.RESTART_ALL_SERVICES)
             .setValidator(new StringLengthValidator(1))
-            .setAccessConstraints(SensitiveTargetAccessConstraintDefinition.SECURITY_REALM_REF)
```
Comment (bstansberry) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpsListenerResourceDefinition.java]: Do we need some sort of SSL_CONTEXT_REF constraint?

This question is OT to this PR even if we do need one. It was just that seeing this bit of the diff sparked the question.

**Comment by bstansberry** [comment]
@kabir looks good to me.

File: undertow/src/main/java/org/wildfly/extension/undertow/HttpsListenerResourceDefinition.java (hunk 1 above)
Comment (darranl) [inline undertow/src/main/java/org/wildfly/extension/undertow/HttpsListenerResourceDefinition.java]: +1 We have a separate RFE to address that one, wanted to keep it separate so we can discuss it as a single unit.  

Will follow up with that one once the management PR is in place as at that point the majority of references will be completed.

**Comment by kabir** [comment]
Retest this please

**Pull request merged**